### PR TITLE
GH-771 TABLE (AUTO CALC ITEMS PER PAGE) AND PAGINATOR UPDATE (ABS-BOT)

### PR DESCRIPTION
**Pull request by mlzummo**
Adds new publish parameter autoCalcIPP
Changes Paginator so that its fixed height
Adds new features to Paginator (jump to page)

Fixes GH-771

### CODE REVIEW

**Comment by mlzummo** [comment]
@GordonSmith Ready for pull

**Comment by mlzummo** [comment]
@GordonSmith @dtsnell4 Please check that this is working and calculating correctly now for the paginator positition http://rawgit.com/mlzummo/Visualization/GH-771_TABLE_PAGINATOR_UPDATE/demos/dermatology.html?src/other/Table

**Comment by dtsnell4** [comment]
@mlzummo There is still a vertical scroll bar when I look at it.   My guess is you may need to subtract the scrollbar width when the horizontal scroll bar is displayed

**Comment by mlzummo** [comment]
fixed

**Comment by mlzummo** [comment]
@GordonSmith Also fixed missing d3 declaration in top of HipieDDL

**Comment by dtsnell4** [comment]
Looks good to me

**Comment by GordonSmith** [comment]
Back to this issue:  
image

**Comment by GordonSmith** [comment]
image

**Comment by jbrundage** [comment]
@mlzummo Toggling **autoCalcIPP** on and off a few times causes the table header to shift towards the left by one column (each time).
https://rawgit.com/mlzummo/Visualization/GH-771_TABLE_PAGINATOR_UPDATE/demos/dermatology.html?src/other/Table&pagination=true&autoCalcIPP=true

image

**Comment by mlzummo** [comment]
@dtsnell4 any thoughts on why thats happening?

**Comment by mlzummo** [comment]
how come this causes that to occur: var thHeight = this.thead.select("th")[0][0].clientHeight; ?

File: src/other/Table.js [hunk 1]
```diff
@@ -146,6 +148,27 @@
...
+            })
+        ;
+        tcell.exit()
+            .remove()
+        ;
+        return tcell;
+    };
+
     Table.prototype._calcRowsPerPage = function(th) {
         if (this._paginator.numItems() === 0) { // only run on first render
```
Comment (mlzummo) [inline src/other/Table.js]: see if we can change to ._renderCount?

**Comment by GordonSmith** [comment]
Closing in favor of:  https://github.com/hpcc-systems/Visualization/pull/815

**Pull request closed without merging**